src/map/utils/fishingutils.cpp

```diff
@@ -65,6 +66,13 @@ void LoadFishingMessages()
     });
 }
 
+void ClearPlayersFishingCatches()
+{
+    Sql_Query(SqlHandle, "update char_vars set value = 0 WHERE varname = 'FishingCatches';");
```

Comment:
Are you missing " and charid = " otherwise you will clear everyone's counters!

Reply:
It's to clear everyone's counter. It's to be called at server midnight. I just realized i didn't commit my changes to time_server.cpp, will fix that.

Reply:
why would you not just check and set this the next time the specific player would need to access (read or write) that variable _anyway_ instead of setting up a new global timer to hit the entire server?

Reply:
I didn't set a new timer, I put it next to "guildutils::UpdateGuildPointsPattern()" call. Shouldn't everything that needs to be midnight reset, be in same place?
How to implement your suggestion?

Reply:
Leave it like it is for now I guess, if its just piggy backing an existing one. It'll actually be more trouble and lines of code than its worth to explain and do to in place check. Honestly, I don't even like our existing implementation but I'm not going to ask you to rewrite it that'd be a little crazy/cruel. 

(if I had my way most of it would move out to global/zone lua's like digging, if only to do away with rebuilding/reimporting when we need to add/fix something)


###### more/slashes/just/because.